pkg/collector/worker/worker.go

```diff
@@ -152,6 +152,21 @@ func (w *Worker) Run() {
 			continue
 		}
 
+		// Use the default sender for the service checks
+		sender, err := w.getDefaultSenderFunc()
+		if err != nil {
+			log.Errorf("Error getting default sender: %v. Not sending status check for %s", err, check)
+		}
+
+		senderShouldCommit := false
+		if sender != nil && w.haAgent.Enabled() && w.haAgent.IsHaIntegration(check.String()) {
+			sender.Count("datadog.agent.ha_agent.integration_runs", 1, "", []string{
+				"integration:" + check.String(),
```

Comment:
Default sender is unsuitable for sending metrics. Aggregation only happens within commit interval, which for the default sender, is unpredictable and can be between 0 and 15 seconds. Multiple instances of integration running concurrently can commit at the same time, sending duplicate metrics to the backend, overwriting each other and causing undercounting.

Please use the telemetry component with the [default telemetry](https://github.com/datadog/datadog-agent/blob/1acf75b49c528945e66419830a9133b5a7d19a16/comp/core/telemetry/options.go#L18-L19) option instead, and move the code in with the rest of checks telemetry [here](https://github.com/datadog/datadog-agent/blob/1acf75b49c528945e66419830a9133b5a7d19a16/pkg/collector/check/stats/stats.go#L190-L203).

Reply:
@vickenty thanks for the suggestion, indeed, that seems to be a much better way.

I updated the code to use telemetry instead: https://github.com/DataDog/datadog-agent/pull/33233/commits/c2f69ca3421701981a3c7bba1d695be7abaf6665

Reply:
@vickenty I would need to bring haagent component to stats.go since we only collect this metric on specific conditions.

It seems that would require a lot of code change to bring haagent to stats.go, not fully sure if it's worth it.

WDYT?

Reply:
@vickenty Updated like this https://github.com/DataDog/datadog-agent/pull/33233/commits/8e41da08a7f39d6203e0a50a0b32434f1f11ffe0 

Can have have a look again?